core/variables.js

```diff
@@ -281,18 +282,9 @@ Blockly.Variables.renameVariable = function(workspace, variable,
       Blockly.Msg.RENAME_VARIABLE_TITLE.replace('%1', variable.name), defaultName,
       function(newName) {
         if (newName) {
-          var newVariable = workspace.getVariable(newName);
-          if (newVariable && newVariable.type != variable.type) {
-            Blockly.alert(Blockly.Msg.VARIABLE_ALREADY_EXISTS_FOR_ANOTHER_TYPE.replace('%1',
-                newName.toLowerCase()).replace('%2', newVariable.type),
-                function() {
-                  promptAndCheckWithAlert(newName);  // Recurse
-                });
-          } else {
-            workspace.renameVariable(variable.name, newName);
-            if (opt_callback) {
-              opt_callback(newName);
-            }
+          workspace.renameVariable(variable.name, newName);
```

Comment:
I think this will break for variables that aren't the '' type.

Reply:
Yes.  Most things are now broken for anything but the '' type, because the field stores only the name right now.  I'll fix it when everything uses an ID.